forgerock-openbanking-uk-aspsp-as/forgerock-openbanking-uk-aspsp-as-gateway/forgerock-openbanking-uk-aspsp-as-gateway-server/src/main/java/com/forgerock/openbanking/aspsp/as/service/TppRegistrationService.java

```diff
@@ -84,102 +81,104 @@ public TppRegistrationService(CryptoApiClient cryptoApiClient,
         }
     }
 
-    public void verifyTPPRegistrationRequestAgainstSSA(OIDCRegistrationRequest oidcRegistrationRequest,
-                                                       JWTClaimsSet ssaClaims) throws OBErrorException, OIDCException {
-        log.debug("Verify TPP registration request matches the SSA request");
-        log.debug("- Verify the software id");
+    /**
+     *
+     * @param registrationRequest
+     * @throws DynamicClientRegistrationException
+     */
+    public void verifyTPPRegistrationRequestAgainstSSA(RegistrationRequest registrationRequest)
+            throws DynamicClientRegistrationException {
+
+        log.debug("verifyTPPRegistrationRequestAgainstSSA() Verify TPP registration request matches the SSA request");
+        log.debug("verifyTPPRegistrationRequestAgainstSSA() Verify the software id");
 
         // The OB Dynamic Client Registration spec says that;
         // If specified, the software_id in the request MUST match the software_id specified in the SSA. ASPSPs can
         // choose to allow multiple registrations for a given software statement. The Software ID must be represented
         // as a Base62 UUID.
         // The cardinality of software_id in the registration request spec os 0..1 meaning this can be null and that's
         // OK.
-        String registrationRequestSoftwareId = oidcRegistrationRequest.getSoftwareId();
-
-        String noSoftwareIdErrorDescription = "Failed to obtain redirect URIs from the software statement";
-        // Throws if no claims available. ssaClaimSetSoftwareId will not be null or empty if it returns
-        String ssaClaimSetSoftwareId = getSsaStringClaim(ssaClaims, OpenBankingConstants.SSAClaims.SOFTWARE_ID,
-                noSoftwareIdErrorDescription);
+        String registrationRequestSoftwareId = registrationRequest.getSoftwareId();
 
         if (StringUtils.isNotBlank(registrationRequestSoftwareId)){
+            // Throws if no claims available. ssaClaimSetSoftwareId will not be null or empty if it returns
+
+            String ssaClaimSetSoftwareId = registrationRequest.getSoftwareIdFromSSA();
             if(!registrationRequestSoftwareId.equals(ssaClaimSetSoftwareId)){
                 String errorDescription = "The software_id in the registration request differs from that in the " +
                         "Software Statement";
-                log.debug(errorDescription + ". ssaClaimSetSoftwareId: '{}', registrationRequestSoftwareId: '{}'",
-                        ssaClaimSetSoftwareId, registrationRequestSoftwareId);
-
-                RegistrationError registrationError = new RegistrationError()
-                        .error(RegistrationError.ErrorEnum.INVALID_CLIENT_METADATA)
-                        .errorDescription(errorDescription);
-                throw new OIDCException(registrationError);
+                log.debug("verifyTPPRegistrationRequestAgainstSSA() " + errorDescription + ". ssaClaimSetSoftwareId: " +
```

Comment:
Could be better log.error instead of log.debug? because contains the errorMessage.

Reply:
we worked out this should be log.info